packages/@lwc/rollup-plugin/src/index.ts

```diff
@@ -170,38 +180,77 @@ export default function lwc(pluginOptions: RollupLwcOptions = {}): Plugin {
+                        importeeExt
+                    );
 
-                const normalizedPath = path.resolve(path.dirname(importer), importee);
-                let absPath = pluginUtils.addExtension(normalizedPath, ext);
+                    // remove query params
+                    const { filename: importeeNormalizedFilename } =
+                        parseDescriptorFromFilePath(importeeAbsPath);
 
-                const { filename } = parseQueryParamsForScopedOption(absPath);
-                absPath = filename; // remove query param
+                    if (
+                        isImplicitHTMLImport(importeeNormalizedFilename, importerFilename) &&
+                        !fs.existsSync(importeeNormalizedFilename)
+                    ) {
+                        return IMPLICIT_DEFAULT_HTML_PATH;
+                    }
 
-                if (isImplicitHTMLImport(absPath, importer) && !fs.existsSync(absPath)) {
-                    return IMPLICIT_DEFAULT_HTML_PATH;
-                }
+                    if (
+                        isImplicitCssImport(importeeNormalizedFilename, importerFilename) &&
+                        !fs.existsSync(importeeNormalizedFilename)
+                    ) {
+                        return IMPLICIT_DEFAULT_CSS_PATH;
+                    }
 
-                if (isImplicitCssImport(absPath, importer) && !fs.existsSync(absPath)) {
-                    return IMPLICIT_DEFAULT_CSS_PATH;
-                }
+                    if ((importeeExt === '.html' || importeeExt === '.css') && importerSpecifier) {
+                        // html and css files should inherit the namespace and name of the module
+                        // importing them
+                        return appendAliasSpecifierQueryParam(importeeAbsPath, importerSpecifier);
+                    }
 
-                return pluginUtils.addExtension(normalizedPath, ext);
-            } else if (importer) {
-                // Could be an import like `import component from 'x/component'`
-                try {
-                    return resolveModule(importee, importer, {
-                        modules,
-                        rootDir,
-                    }).entry;
-                } catch (err: any) {
-                    if (err && err.code !== 'NO_LWC_MODULE_FOUND') {
-                        throw err;
+                    return importeeAbsPath;
+                } else {
+                    // Could be an import like `import component from 'x/component'`
+                    try {
+                        const { entry, specifier, type } = resolveModule(importee, importer, {
+                            modules,
+                            rootDir,
+                        });
+
+                        if (type === RegistryType.alias) {
+                            // specifier must be in in namespace/name format
+                            const hasValidSpecifier = specifier.split('/').length > 1;
```

Comment:
```suggestion
                            const hasValidSpecifier = specifier.split('/').filter(Boolean).length === 2;
```

Strings like `foo/` and `/bar` are not valid specifiers. Also I'm pretty sure `foo/bar/baz` is not valid.